File: Solution/CoMentor/Integrated/dotnetintegrated/backend/Repositories/Interfaces/IQuestionsRepository.cs
using backend.DTOs;
using System.Threading.Tasks;

namespace backend.Repositories.Interfaces
{
    public interface IQuestionsRepository
    {
        Task<QuestionsDto> GetQuestionDetailsWithAnswer(int questionId);
        Task<IEnumerable<AssessmentQuestionAnswersDto>> GetStudentAssessmentQuestionsResultAsync(int assessmentId, int studentId);
        Task<QuestionDetailsDto> GetQuestionDetails(int questionId);
    }
  
}

File: Solution/CoMentor/Integrated/javaintegrated/Backend/TFLComentor/src/main/java/com/transflower/tflcomentor/skilltaxonomy/repository/IQuestionsRepository.java
package com.transflower.tflcomentor.skilltaxonomy.repository;

import org.springframework.stereotype.Repository;

import com.transflower.tflcomentor.skilltaxonomy.dto.QuestionDetailsDto;
import com.transflower.tflcomentor.skilltaxonomy.dto.StudentAssessmentDto;

@Repository
public interface IQuestionsRepository {
    
    QuestionDetailsDto questionDetailsWithAns(int id);
    
    StudentAssessmentDto getStudentAssessmentQuestionsResultAsync(int assessmentId, int studentId);
    
    QuestionDetailsDto getQuestionDetailsWithAnswer(int questionId);
    
    QuestionDetailsDto getQuestionDetails(int questionId);
}
